internal/vds/vds.cpp

```diff
@@ -450,13 +365,22 @@ struct response fetch_slice_metadata(
      * would be quite suprising for people that use this API in conjunction
      * with the OpenVDS library.
      */
-    std::vector< int > dims;
-    for (int i = 0; i < 3; ++i) {
-        if (i == vdim) continue;
-        dims.push_back(i);
+    const Axis inline_axis(axis_name::INLINE, layout);
+    const Axis crossline_axis(axis_name::CROSSLINE, layout);
+    const Axis sample_axis(axis_name::SAMPLE, layout);
+
+    if (ax == axis_name::I or ax == axis_name::INLINE) {
```

Comment:
If we remove axis name from construction (and implement comparison operators? interesting if we can assert that axis describes the same vds file or not), could we then have just 
`if (ax == inline_axis)...`

Also as the logic got changed, question for future: do we always want to return axes in the order you established, or do we want the previous logic?
In previous logic we return whatever order there is in the file.

Relevant for other question: would we assert that files have order sample-crossline-inline or would we go further and read the order from volumeDataLayout?
If first, you current solution looks more readable.
If later, we have to decide whether we always return in current order, or have to switch here to solution where we do not assume the order.

Reply:
1. We can definitely implement some comparison operator. I guess we could, in the "worst" case base the comparison on the name string defined available in the VDS. So something like
   ```c++
   bool operator==(const enum axis_name an, const Axis& ax){
     if ((an == axis_name::I) or (an == axis_name::INLINE)) {
       return (ax.get_name() == "I") or (ax.get_name() == "Inline");
     } else if ((an == axis_name::J) or (an == axis_name::CROSSLINE)) {
       return (ax.get_name() == "J") or (ax.get_name() == "Crossline");
     } else {
        return (ax.get_name() == "Depth")
                or (ax.get_name() == "Time")
                or (ax.get_name() == "Sample");
     }
   }
   ```
2. If we want to be future proof, we can think of a way to restore the old logic.
3. The change here, for now, assumes that we have guaranteed order Sample, Crossline, Inline in the VDS. The way I understood it was that we want to allow for arbitrary order of axes, but not during the refactoring in order to keep the noise/number of changes low.